### ISSUE TITLE
[SDK 2.0] Support Event sheet classes 

### ISSUE THREAD

**Issue by armandoalonso** [Scirra/Construct-feature-requests#270]
### Reviewed guidelines

- [X] I have read and understand the suggestion guidelines

### Checked for duplicate suggestions

- [X] I checked for existing similar suggestions

### Summary

Currently on the SDK v1.0 api Event sheet API is supported and accessible
https://www.construct.net/en/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventblock
https://www.construct.net/en/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventsheetmanager
https://www.construct.net/en/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventstack
https://www.construct.net/en/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventstackframe

### Possible workarounds or alternatives

with the new sdk being limited by encapsulation, i don't think there is a workaround with bypassing encapsulation

### Proposed solution

expose the event sheet api in the IRuntime

IRuntime.GetEventSheetManager()
Return the [EventSheetManager](https://www.construct.net/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventsheetmanager) that handles the event system.

IRuntime.GetEventStack()
Return the [EventStack](https://www.construct.net/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventstack). This is a shorthand for GetEventSheetManager().GetEventStack().

IRuntime.GetCurrentEventStackFrame()
Return the current [EventStackFrame](https://www.construct.net/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventstackframe). This is a shorthand for GetEventSheetManager().GetCurrentEventStackFrame().

IRuntime.GetCurrentEvent()
Return the current [EventBlock](https://www.construct.net/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventblock). This is a shorthand for GetEventSheetManager().GetCurrentEvent().

### Why is this idea important?

Addons need this information to properly implemtent thier own loops detailed here:

https://www.construct.net/en/make-games/manuals/addon-sdk/runtime-reference/event-sheet-classes/eventblock

```
MyLoopingCondition()
{
	// Get necessary references
	const runtime = this._runtime;
	const eventSheetManager = runtime.GetEventSheetManager();
	const currentEvent = runtime.GetCurrentEvent();
	const solModifiers = currentEvent.GetSolModifiers();
	const eventStack = runtime.GetEventStack();

	// Get current stack frame and push new one
	const oldFrame = eventStack.GetCurrentStackFrame();
	const newFrame = eventStack.Push(currentEvent);

	for (const item of myArray)
	{
		// optionally update state here 

		// Push a copy of the current SOL
		eventSheetManager.PushCopySol(solModifiers);

		// Retrigger the current event, running a single loop iteration
		currentEvent.Retrigger(oldFrame, newFrame);

		// Pop the current SOL
		eventSheetManager.PopSol(solModifiers);
	}

	// Pop the event stack frame
	eventStack.Pop();

	// Return false since event already executed
	return false;
}
```

**Comment by LuanHimmlisch** [comment]
@armandoalonso yeah, in theory any of the "SDK v2.0" feature request are actually scripting interface feature requests

**Comment by skymen** [comment]
> if this also becomes available from the scripting api as a consequence of the runtime being shared that would also be pretty neat. it would allow the scripting api to mess with the flow of events.

Agreed, this would be very sick IMO

**Comment by AshleyScirra** [comment]
If the motivation is to make looping conditions, this can be achieved in SDKv2 using [ILoopingConditionContext](https://www.construct.net/en/make-games/manuals/construct-3/scripting/scripting-reference/addon-sdk-interfaces/iloopingconditioncontext).
